@xen-orchestra/backups/writers/DeltaBackupWriter.js

```diff
@@ -145,7 +145,7 @@ exports.DeltaBackupWriter = class DeltaBackupWriter extends MixinBackupWriter(Ab
               // don't do delta for it
               vdi.uuid
             : vdi.$snapshot_of$uuid
-        }/${basename}.vhd`
+        }/${adapter.getVhdFileName(basename)}`
```

Comment:
Why not keep this problematic out of the `DeltaBackupWriter` and let `RemoteAdapter#outputStream()` handle it directly?

Reply:
you're right, it seems like a problematic for RemoteAdapter.  but the Vhd file must be in the metadata in the end. Should the remoteAdapter return the path ?

Reply:
I missed that indeed, maybe it should indeed returns the VHD path.

Reply:
outputStream is also used in fullBackupWriter. I think I will add another method `outputVhd` or `writeVhd` that will do its magic and call the original `outputStream` if needed